### ISSUE TITLE
adonis and promises?

### ISSUE THREAD

**Issue by callen5914** [adonisjs/adonis-framework#298]
Using the following code: 

    'use strict'

    const Particle = require('particle-api-js')
    const particle = new Particle();

    class DeviceController {
	* index(request, response){

		const token = request.cookie('token')

		const devicesPr = particle.listDevices({ auth: token })

		devicesPr.then(
  			function(devices){
    			//console.log('Devices: ', devices.body);
  			},
  			function(err) {
    			console.log('List devices call failed: ', err);
  			}
  		).then(
  			yield response.sendView('devices', {devices})
  		)
	
	 }
    }

    module.exports = DeviceController`

I am not able to get the device listing I seek.

Here is the output in the console:

Devices:  [ { id: '3b003c001847343337363432',
    name: 'Photon3',
    last_app: null,
    last_ip_address: '68.102.142.217',
    last_heard: '2016-09-25T17:05:18.231Z',
    product_id: 6,
    connected: false,
    platform_id: 6,
    cellular: false,
    status: 'normal' },
  { id: '210023000247343138333038',
    name: 'Photon1',
    last_app: null,
    last_ip_address: '68.102.142.217',
    last_heard: '2016-09-25T16:45:05.168Z',
    product_id: 6,
    connected: true,
    platform_id: 6,
    cellular: false,
    status: 'normal',
    pinned_build_target: '0.4.9' },
  { id: '3a003b001447343432313031',
    name: 'Photon2',
    last_app: null,
    last_ip_address: '68.102.142.217',
    last_heard: '2016-09-26T12:43:44.985Z',
    product_id: 6,
    connected: true,
    platform_id: 6,
    cellular: false,
    status: 'normal',
    pinned_build_target: '0.5.2' } ]

So basically I wanted to use this data to list all the devices and related info in the view using the njk templating tags. I have been fight with this for about 6 hrs. Now we are quickly reaching the definition for insanity (do the same thing over and over again with the same unsuccessful results). Any thoughts?

**Comment by thetutlage** [comment]
You can simply `yield` your promises.

```javascript
const devicesPr = yield particle.listDevices({ auth: token })
yield response.sendView('devices', {devices: devicesPr.body})
```

**Comment by callen5914** [comment]
TypeError: devicesPr.then is not a function<br> &nbsp; &nbsp;at DeviceController.index (/Users/claytonallen/Projects/klaxon/app/Http/Controllers/DeviceController.js:13:13)<br> &nbsp; &nbsp;at next (native)<br> &nbsp; &nbsp;at /Users/claytonallen/Projects/klaxon/node_modules/adonis-framework/src/Middleware/index.js:207:20<br> &nbsp; &nbsp;at next (native)<br> &nbsp; &nbsp;at onFulfilled (/Users/claytonallen/Projects/klaxon/node_modules/co/index.js:65:19)<br> &nbsp; &nbsp;at run (/Users/claytonallen/Projects/klaxon/node_modules/core-js/library/modules/es6.promise.js:87:22)<br> &nbsp; &nbsp;at /Users/claytonallen/Projects/klaxon/node_modules/core-js/library/modules/es6.promise.js:100:28<br> &nbsp; &nbsp;at flush (/Users/claytonallen/Projects/klaxon/node_modules/core-js/library/modules/_microtask.js:18:9)<br> &nbsp; &nbsp;at nextTickCallbackWith0Args (node.js:420:9)<br> &nbsp; &nbsp;at process._tickCallback (node.js:349:13)

**Comment by callen5914** [comment]
That was error copied to clipboard

**Comment by thetutlage** [comment]
Does `listDevices` returns a Promise?

**Comment by callen5914** [comment]
How might I go about finding that answer?

**Comment by thetutlage** [comment]
Can u share your `listDevices` method?

**Comment by callen5914** [comment]
Here's a console log of devicesPr:

Promise {
  _c: 
   [ { promise: [Object],
       resolve: [Function],
       reject: [Function],
       ok: [Function],
       fail: [Function],
       domain: null } ],
  _a: undefined,
  _s: 0,
  _d: false,
  _v: undefined,
  _h: 0,
  _n: false }

**Comment by callen5914** [comment]
This is pulled directly from Particle.io:

var token; // from result of particle.login
var devicesPr = particle.listDevices({ auth: token });

devicesPr.then(
  function(devices){
    console.log('Devices: ', devices);
  },
  function(err) {
    console.log('List devices call failed: ', err);
  }
);

**Comment by callen5914** [comment]
It's their API to list all devices

**Comment by thetutlage** [comment]
Promises should work fine without any issue and checking 3rd party API's is outside the scope of this repo. Closing the issue, please feel free to report any future bugs
